### PR TITLE
jetbrains.jcef: 611 -> 654 

### PR DESCRIPTION
**Pull request by niknetniko**
###### Description of changes

With #223149, most Jetbrains IDEs have been moved to 231.*. These IDEs need a newer version of JCEF to work properly (e.g. Markdown previews). This PR switches the JCEF version of the Jetbrains JDK to the [231 branch](https://github.com/JetBrains/jcef/tree/231).

I've also tested Goland (which is not yet on 231.*) and it also works.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

File: pkgs/development/compilers/jetbrains-jdk/jcef.nix [hunk 1]
```diff
@@ -76,8 +76,8 @@ buildType = if debugBuild then "Debug" else "Release";
 
 in stdenv.mkDerivation rec {
   name = "jcef-jetbrains";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/jetbrains-jdk/jcef.nix:78]:
```suggestion
  pname = "jcef-jetbrains";
```

File: pkgs/development/compilers/jetbrains-jdk/jcef.nix [hunk 2]
```diff
@@ -76,8 +76,8 @@ buildType = if debugBuild then "Debug" else "Release";
...
-  rev = "153d40c761a25a745d7ebf0ee3a024bbc2c840b5";
-  commit-num = "611";  # Run `git rev-list --count HEAD`
+  rev = "3dfde2a70f1f914c6a84ba967123a0e38f51053f";
+  commit-num = "654";  # Run `git rev-list --count HEAD`
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/jetbrains-jdk/jcef.nix:80]:
```suggestion
  version = "654";  # Run `git rev-list --count HEAD`
```
How about we use the commit number as the version? Before the package did not have one at all. This also solves the question how to write the commit message

**Review by SebTM — APPROVED** [review]
Using it for quite some time and works fine for me :)

**Comment by SuperSandro2000** [comment]
@ofborg build jetbrains.jcef

**Pull request merged**